### PR TITLE
Modify aten._log_softmax op decomposition for numerical stability.

### PR DESCRIPTION
**Pull request by pashu123**
`aten.log_softmax` is decomposed to be more numerically stable.

### CODE REVIEW

**Review by cathyzhyi — CHANGES_REQUESTED** [review]
Thanks for being careful about numerical stability!

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -468,25 +496,31 @@ class DecomposeAtenLogSoftmaxIntOp
     if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
       return rewriter.notifyMatchFailure(op, "Only support floating type");
 
-    // softmax(x, dim)
-    Value softmax = rewriter.create<AtenSoftmaxIntOp>(loc, tensorType, self,
-                                                      dim, op.dtype());
-    rewriter.replaceOpWithNewOp<AtenLogOp>(op, op.getType(), softmax);
+    Value logSoftmax = getLogSoftmaxResult(op, rewriter);
```
Comment (cathyzhyi) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:499]: nit: `logSoftmax` var is not needed.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 2]
```diff
@@ -468,25 +496,31 @@ class DecomposeAtenLogSoftmaxIntOp
...
+    if (!matchPattern(op.half_to_float(), m_TorchConstantBool(&halfToFloat)))
+      return rewriter.notifyMatchFailure(
+          op, "Expected a boolean value for half_to_float");
+
+    // Currently, setting `halfToFloat` is not supported as the E2E testing for
+    // the same is not present on CPU.
+    if (halfToFloat)
+      return rewriter.notifyMatchFailure(
+          op, "halfToFloat is currently not supported.");
+    Value _logSoftmax = getLogSoftmaxResult(op, rewriter);
```
Comment (cathyzhyi) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:522]: nit: `_logSoftmax` is not needed

**Review by cathyzhyi — APPROVED** [review]
just 2 nits.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp (hunk 1 above)
Comment (ramiro050) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:499]: A check is needed to make sure the return of `getLogSoftmaxResult` is not `nullptr`. The same applies for the `Aten_LogSoftmaxOp` version

Comment (pashu123) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:499]: Done, thanks for the catch.

**Pull request merged**